shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm

```diff
@@ -153,6 +155,37 @@ - (void)setSystemChromePreferredOrientations:(NSArray*)orientations {
                   userInfo:@{@(kOrientationUpdateNotificationKey) : @(mask)}];
 }
 
+- (NSArray*)getSystemChromePreferredOrientations {
+  NSMutableArray* orientations = [[NSMutableArray alloc] init];
+  UIViewController* engineViewController = [_engine.get() viewController];
+  UIInterfaceOrientationMask supportedOrientations =
+      engineViewController.supportedInterfaceOrientations;
+  const int possibleOrientationsLength = 4;
+  UIInterfaceOrientationMask possibleOrientations[possibleOrientationsLength] = {
+      UIInterfaceOrientationMaskPortrait, UIInterfaceOrientationMaskPortraitUpsideDown,
+      UIInterfaceOrientationMaskLandscapeLeft, UIInterfaceOrientationMaskLandscapeRight};
+  for (int i = 0; i < possibleOrientationsLength; i++) {
+    unsigned long mask = possibleOrientations[i];
+    unsigned long orientation = supportedOrientations & mask;
+    switch (orientation) {
+      case UIInterfaceOrientationMaskPortrait:
+        [orientations addObject:@"DeviceOrientation.portraitUp"];
+        break;
+      case UIInterfaceOrientationMaskPortraitUpsideDown:
+        [orientations addObject:@"DeviceOrientation.portraitDown"];
+        break;
+      case UIInterfaceOrientationMaskLandscapeLeft:
+        [orientations addObject:@"DeviceOrientation.landscapeLeft"];
+        break;
+      case UIInterfaceOrientationMaskLandscapeRight:
+        [orientations addObject:@"DeviceOrientation.landscapeRight"];
+        break;
+    }
+  }
```

Comment:
Since you already need to handle each case in the switch anyway, how about avoiding the loop and c-style array and length:
```suggestion
  if (supportedOrientations & UIInterfaceOrientationMaskPortrait) {
    [orientations addObject:@"DeviceOrientation.portraitUp"];
  }
  if (supportedOrientations & UIInterfaceOrientationMaskPortraitUpsideDown) {
    [orientations addObject:@"DeviceOrientation.portraitDown"];
  }
  if (supportedOrientations & UIInterfaceOrientationMaskLandscapeLeft) {
    [orientations addObject:@"DeviceOrientation.landscapeLeft"];
  }
  if (supportedOrientations & UIInterfaceOrientationMaskLandscapeRight) {
    [orientations addObject:@"DeviceOrientation.landscapeRight"];
  }
```